.github/workflows/pre-commit-update.yml

```
@@ -0,0 +1,34 @@
+name: Pre-commit auto-update
+on:
+  schedule:
+    # Run on mondays at midnight
+    - cron: "0 0 * * 1"
+
+jobs:
+  auto-update:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.1.2
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
```

```suggestion
        uses: actions/setup-python@v5.0.0
```
It would be good to pin `setup-python` we do the same in other workflows:
https://github.com/home-assistant-libs/aioshelly/blob/497b7a44cc954198bfba4e98a62c163bf5ffc841/.github/workflows/pythonpublish.yml#L17